### DISCUSSION TITLE
How to draw string/text on core.Canvas ?

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1361] [General]
I cant find any function to draw text on a canvas ??

**Comment by kkoreilly** [comment]
It is possible to draw text on a canvas as shown in the example below. I will make the API simpler for this and add documentation to the website soon.

```go
core.NewCanvas(b).SetDraw(func(pc *paint.Context) {
	txt := &paint.Text{}
	fr := &styles.FontRender{}
	fr.Defaults()
	stx := &styles.Text{}
	stx.Defaults()
	txt.SetHTML("Hello, world", fr, stx, &pc.UnitContext, nil)
	txt.Layout(stx, fr, &pc.UnitContext, math32.Vec2(50, 50))
	txt.Render(pc, math32.Vec2(10, 10))
})
```

**Reply by heejit** [reply]
It is working but I am not able to understand api properly.  

Simpler api can help  -> paint.Context.DrawText(x, y, text, style) when need we can use extended one how you explain.

Framework is really good but require documentation to use it properly.  Something like How-to section in doc website can help.

**Reply by kkoreilly** [reply]
Yes, I will make the API simpler and add instructions on how to use it to the doc website soon.
